front/lib/api/assistant/jit_actions.ts

```diff
@@ -100,8 +99,8 @@ export async function renderConversationForModelJIT({
           actions: [],
         }) satisfies (typeof steps)[number];
 
-      for (const action of actions) {
-        const stepIndex = action.step;
+      for (let stepIndex = 0; stepIndex < actions.length; stepIndex++) {
+        const action = actions[stepIndex];
```

Comment:
so now the action.step do not necessary match the stepIndex ?

Reply:
Yes this one still need a bit of work I was a bit too rapid to change the behavior here